aggregator/src/task.rs

```diff
@@ -373,9 +376,89 @@ struct SerializedTask {
     input_share_aad_public_share_length_prefix: bool,
 }
 
+impl SerializedTask {
+    /// Returns the task ID, if one is set.
+    pub fn task_id(&self) -> Result<Option<TaskId>, Error> {
+        Ok(self
+            .task_id
+            .as_deref()
+            .map(TaskId::from_base64_url_no_padding)
+            .transpose()?)
+    }
+
+    /// Randomly generates and fills values for the following fields if they are not set in the
+    /// [`SerializedTask`]
+    ///
+    /// - Task ID
+    /// - VDAF verify keys (only one key is generated)
+    /// - Aggregator authentication tokens (only one token is generated)
+    /// - Collector authentication tokens (only one token is generated and only if the task's role
+    ///   leader)
+    /// - The aggregator's HPKE keypair (only one keypair is generated and a random HPKE config ID
+    ///   that does not occur in `claimed_hpke_config_ids` is used)
+    pub fn generate_missing_fields(&mut self, claimed_hpke_config_ids: &[HpkeConfigId]) {
+        if self.task_id.is_none() {
+            let task_id: TaskId = random();
+            self.task_id = Some(base64::encode_config(task_id.as_ref(), URL_SAFE_NO_PAD));
+        }
+
+        if self.vdaf_verify_keys.is_empty() {
+            let vdaf_verify_key = SecretBytes::new(
+                thread_rng()
+                    .sample_iter(Standard)
+                    .take(self.vdaf.verify_key_length())
+                    .collect(),
+            );
+
+            self.vdaf_verify_keys = vec![base64::encode_config(
```

Comment:
nit: prefer `Vec::from` for new code (as long as Clippy doesn't complain -- I think David found one corner case where Clippy reports that `vec!` is still more efficient for now)